From 475386e7551e74b730936cb8fbeb32d75bb6b182 Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey@solus-project.com>
Date: Sat, 15 Apr 2017 02:21:02 +0100
Subject: [PATCH 2/5] configfile: Unjank our flags

This returns the original ordering of ldflags to make it easier to track
them over time. Additionally we return to -fPIC by default, as Solus uses
ASLR and without PIC we cannot benefit.

Signed-off-by: Ikey Doherty <ikey@solus-project.com>
---
 pisi/configfile.py | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/pisi/configfile.py b/pisi/configfile.py
index 9b2968b..52cc039 100644
--- a/pisi/configfile.py
+++ b/pisi/configfile.py
@@ -85,9 +85,9 @@ class BuildDefaults:
     jobs = "-j5"
     generateDebug = False
     enableSandbox = False # Dropping sandbox support soon
-    cflags = "-mtune=generic -march=x86-64 -O2 -g -feliminate-unused-debug-types -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector --param=ssp-buffer-size=32 -Wformat -Wformat-security -fasynchronous-unwind-tables -ftree-loop-vectorize -ftree-vectorize -fopt-info-vec -fno-omit-frame-pointer -Wno-error -Wp,-D_REENTRANT"
-    cxxflags = "-mtune=generic -march=x86-64 -O2 -g -feliminate-unused-debug-types -pipe -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector --param=ssp-buffer-size=32 -fasynchronous-unwind-tables -ftree-loop-vectorize -ftree-vectorize -fopt-info-vec -fno-omit-frame-pointer -Wno-error -Wp,-D_REENTRANT"
-    ldflags = "-Wl,--copy-dt-needed-entries -Wl,-z -Wl,now -Wl,-z -Wl,relro -Wl,-z,max-page-size=0x1000"
+    cflags = "-mtune=generic -march=x86-64 -O2 -g -feliminate-unused-debug-types -pipe -fPIC -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector --param=ssp-buffer-size=32 -Wformat -Wformat-security -fasynchronous-unwind-tables -ftree-loop-vectorize -ftree-vectorize -fopt-info-vec -fno-omit-frame-pointer -Wno-error -Wp,-D_REENTRANT"
+    cxxflags = "-mtune=generic -march=x86-64 -O2 -g -feliminate-unused-debug-types -pipe -fPIC -Wall -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector --param=ssp-buffer-size=32 -fasynchronous-unwind-tables -ftree-loop-vectorize -ftree-vectorize -fopt-info-vec -fno-omit-frame-pointer -Wno-error -Wp,-D_REENTRANT"
+    ldflags = "-Wl,--copy-dt-needed-entries -Wl,-z,relro -Wl,-z,now -Wl,-z,max-page-size=0x1000"
     buildhelper = "ccache"
     compressionlevel = 1
     fallback = "https://www.solus-project.com/source"
-- 
2.12.2

